### DISCUSSION TITLE
More drag and drop type capability

### DISCUSSION THREAD

**Discussion by GullyGully** [freeplane/freeplane#1236] [Ideas]
Hi. I've been mind mapping on paper and PC since the late 1980s. Have been mostly using MindMangaer (mindjet) for the past 20 years but also tinker with Xmind nowadays. 

So my recent experience with Freeplane is that it's not particularly nimble when it comes to drag and drop functionality. And it would also be nice to be able to insert new branches via the insert and enter key. Are there any plans to address these areas?  Thanks

**Comment by GullyGully** [comment]
Perhaps I was a little to hasty to comment. I have now discovered the TAB key function for insert and CTLR Up/Down will move branches around. Just a little  counter intuitive from what I am used to.

**Comment by sifran-github** [comment]
I change my keys to:
insert new node: insert (i support returning insert to be the default, like it was before).
move nodes: alt+shift+arrows instead of ctrl+arrows.

**Reply by EdoFro** [reply]
The problem with the Insert Key is that in many Laptops it's not easy available. For example in mine, you have to use the Function key with the F12 key to get the Insert. In my previous Laptop I had to use the Function Key with the Del key, so any time I pressed it wrong it deleted the node I had selected.

There are other applications where Tab is used to add information horizontally end Enter vertically, So there are reasons for both keys.

I don't think this has to be discussed again.

Maybe the real problem is why new users have to come here to realize that for inserting new nodes they can use the Tab key and that if they want to change that, they can do it easily by making ctrl click on the menu command.

**Reply by sifran-github** [reply]
> The problem with the Insert Key is that in many Laptops it's not easy available. 

I agree that Insert Key is not 100% perfect.
But in my mind Tab is more about... "move to next element" and not "insert new element". i'm talking in general, not only in freeplane.

> I don't think this has to be discussed again.

I agree, but...

> Maybe the real problem is why new users have to come here to realize that for inserting new nodes they can use the Tab key

This is "educating the users"... but in my opinion, if people are complaining about it, and are intuitively try the Insert Key, then it does show that there is an issue... They are used to it from other places (like myself).

> and that if they want to change that, they can do it easily by making ctrl click on the menu command.

True, but new users don't know it, and for new users there are many basic features that are more important than customizing the keys...

**Reply by sifran-github** [reply]
With that being said:
Personally, I don't care if it stays Tab. I am changing it to Insert anyway and I'm happy with it. I change other keys as well.

Just... if more users are complaining about it... then its worth thinking about changing it back.

**Reply by dpolivaev** [reply]
I don't like coming back to old discussions without having new arguments.
I decided to establish a TAB instead of an INSERT key for new nodes because nowadays some keyboards do not have INSERT keys.
I can not assign more than one hot key to a menu item because only one key is indicated.
If new users are not able to check at the menu items which hot keys are available and not able to google how to change a hot key on Freeplane I do not think there is anything I can do to help them.

<img width="962" alt="google" src="https://github.com/freeplane/freeplane/assets/1558052/82f35c98-0806-4d36-87c7-d3c49c2021cd">

**Reply by GullyGully** [reply]
Sorry for the commotion. I am now adjusting to the new input controls. Each time I hit insert it reminds me to hit Tab instead ;-) Keep up the great work.
